.github/workflows/build-release.yml

```diff
@@ -46,12 +46,13 @@ jobs:
       - name: 'Set environment variables: version number'
         run: |
           BITCOIN_PATCHED_VERSION=$(grep -oP "(?<=^CMAKE_PROJECT_VERSION:STATIC=).+(?=)" build/CMakeCache.txt)
-          BRANCH_NAME=${GITHUB_REF#refs/heads/}
-          # Escape '/', '\' and space characters
-          ESCAPED_BRANCH_NAME=$(echo "$BRANCH_NAME" | sed 's/[\/\\ ]/\\&/g')
-          if [ "$ESCAPED_BRANCH_NAME" != "master" ]; then
-            BITCOIN_PATCHED_VERSION="${ESCAPED_BRANCH_NAME}-${BITCOIN_PATCHED_VERSION}"
+          BRANCH_NAME=$(echo $GITHUB_REF | sed 's/refs\/heads\///' | sed 's/refs\/pull\/\([0-9]*\)\/merge/pr-\1/')
```

Comment:
IIUC, this would potentially cause collisions, ie.
`branch-name` and `branch/name`.

Maybe append a hash of the original branch name, if the original branch name contains forbidden characters
https://github.com/actions/toolkit/blob/main/packages/artifact/docs/faq.md#supported-characters

Reply:
If we want to avoid collisions, we need to do something else, eg branchname-commithash-version.

It’s ok to have multiple branches with the exact same name, if the previous one was deleted from GitHub.

Reply:
Collisions on the same branch are intentional. The branch name is in there so that different branches of this repo generate artifacts with different names.

Reply:
Yes.

You can still have collisions though. If you create a branch today called signet, merge it, delete it, and then one month down the line create a branch called signet, you’re going to have a collision

Reply:
We are expecting to have multiple branches in this repo that should create releases, not just `master`. This repo is intended to be used for all forks of Bitcoin that we maintain

Reply:
OK, I can’t quite figure out how to be 100% collission resistant then.

Is it okay to have collisions on very similar branch names?

Reply:
Are you 100% positive all PRs should be uploaded to releases.drivechain.info? Torkels OP_RETURN PR would be uploaded. Same for all future changes and bug-fixes. One unique upload for each PR. If it is the artifacts we want, that’s uploaded for each PR, and can be downloaded from GitHub actions.

If that’s wanted, lets go, but I don’t know how to make it 100% collision resistant.

If we want that, is it okay with minor collissions? Or do you have any idea how we can achieve it?